src/ray/raylet/main.cc

```diff
@@ -37,6 +38,17 @@ int main(int argc, char *argv[]) {
   const std::string java_worker_command = std::string(argv[13]);
   const std::string redis_password = (argc >= 15 ? std::string(argv[14]) : "");
   const std::string temp_dir = (argc >= 16 ? std::string(argv[15]) : "/tmp/ray");
+  const std::string disable_stats_str(argc >= 17 ? std::string(argv[16]) : "false");
+  const bool disable_stats = ("true" == disable_stats_str);
```

Comment:
```suggestion
  bool disable_stats = ("true" == disable_stats_str);
```

I think `bool` shouldn't be `const`.

Reply:
Sorry @robertnishihara, I can't see why `bool` shouldn't be `const`, Could u plz explain a little more?

Reply:
Ok I just looked at the Google C++ style guide, and here is what is says


see https://google.github.io/styleguide/cppguide.html#Use_of_const

> Use const whenever it makes sense. With C++11, constexpr is a better choice for some uses of const.
>
> **Definition:** Declared variables and parameters can be preceded by the keyword const to indicate the variables are not changed (e.g., const int foo). Class functions can have the const qualifier to indicate the function does not change the state of the class member variables (e.g., `class Foo { int Bar(char c) const; };`).
>
> **Pros:** Easier for people to understand how variables are being used. Allows the compiler to do better type checking, and, conceivably, generate better code. Helps people convince themselves of program correctness because they know the functions they call are limited in how they can modify your variables. Helps people know what functions are safe to use without locks in multi-threaded programs.
>
> **Cons:** const is viral: if you pass a const variable to a function, that function must have const in its prototype (or the variable will need a const_cast). This can be a particular problem when calling library functions.
>
> **Decision:** const variables, data members, methods and arguments add a level of compile-time type checking; it is better to detect errors as soon as possible. Therefore we strongly recommend that you use const whenever it makes sense to do so:
>   - If a function guarantees that it will not modify an argument passed by reference or by pointer, the corresponding function parameter should be a reference-to-const (const T&) or pointer-to-const (const T*), respectively.
>   - Declare methods to be const whenever possible. Accessors should almost always be const. Other methods should be const if they do not modify any data members, do not call any non-const methods, and do not return a non-const pointer or non-const reference to a data member.
>   - Consider making data members const whenever they do not need to be modified after construction.
> The mutable keyword is allowed but is unsafe when used with threads, so thread safety should be carefully considered first.

There are two reasons for using `const` I believe, performance and clarity. The performance argument doesn't apply in the bool/int/float case, but the clarity argument does. In the past we haven't been using `const` for this case, but maybe we should (at least for local variables). I think it may not make sense to do `const int f() { return 1; }` or `void f(const int x) { return; }`. However, I'm not 100% sure.

So, using that logic, we should start using `const` for local variables that are not meant to be modified, even if they can't really be mutated. So we should leave the `const` in this case.

Reply:
I think it's a good practice to use `const` for local variables that won't mutate. One reason is clarity, other is that it can avoid accidentally modifying the variable.

Reply:
Sounds good, let's do this.